server/webhook.go

```diff
@@ -21,9 +20,18 @@ const (
 )
 
 func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
-	params := mux.Vars(r)
-
-	if subtle.ConstantTimeCompare([]byte(p.configuration.WebhookSecret), []byte(params["webhooksecret"])) == 0 {
+	secret := r.URL.Query().Get("secret")
+	config := p.getConfiguration()
+	if err := config.IsValid(); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Sorry @sibasankar-demansol, when I said:

> We should put a check in `config.IsValid` to ensure it's set there as well.

I meant that we should edit the `IsValid` function and ensure the webhook secret is set there. We don't need to call `config.IsValid` in this function.

Reply:
Sorry my bad , have  made the required changes.
Please let me know if any thing need to be changed